xgcm/grid.py

```diff
@@ -1150,12 +1150,38 @@ def __init__(
                         f"Input `{dim}` (for the `{pos}` position on axis `{axis}`) is not a dimension in the input datasets `ds`."
                     )
 
+        # Convert all inputs to axes-kwarg mappings
+        # Parse axis properties
+        boundary = self._as_axis_kwarg_mapping(boundary, axes=all_axes)
+        fill_value = self._as_axis_kwarg_mapping(fill_value, axes=all_axes)
+
+        # this case does only apply to periodic. Since we plan on deprecating it soon
+        # handle it here
+        if isinstance(periodic, list):
+            periodic = {axname: True for axname in periodic}
+        periodic = self._as_axis_kwarg_mapping(periodic, axes=all_axes)
+
+        # Set properties on
+
+        # Populate axes. Much of this is just for backward compatibility. I think we are better of
+        # getting completely rid of the axis object and storing/getting info from the grid object properties
+        # (all of which are axis-value mapping).
+        self.boundary = boundary
+        self.fill_value = fill_value
+        self.periodic = periodic
+        # TODO we probably want to properly define these as class properties with setter/getter?
+
         self.axes = OrderedDict()
         for axis_name in all_axes:
-            try:
-                is_periodic = axis_name in periodic
-            except TypeError:
-                is_periodic = periodic
```

Comment:
I'm pleased you got rid of this catching TypeError pattern

Reply:
Eventually we will be able to remove that whole block below too. For now I opted for the least intrusive change.